shpc/client/__init__.py

```diff
@@ -100,6 +100,7 @@ def get_parser():
     install.add_argument(
         "install_recipe",
         help="recipe to install\nshpc install python\nshpc install python:3.9.5-alpine",
+        nargs="+",
     )
```

Comment:
I don't like that `nargs="+"` will take every remaining argument into the list. Maybe even the `--` options thus negating `ensure_no_extra` ?
I'd rather add an extra argument that has `nargs="?"` for the sif only, see below

```suggestion
    install.add_argument(
        "install_recipe",
        help="recipe to install\nshpc install python\nshpc install python:3.9.5-alpine",
    )
    install.add_argument(
        "container_image",
        help="path to an existing container image for this software",
        nargs="?",
    )
```